### ISSUE TITLE
New contentrec errors 0.16

### ISSUE THREAD

**Issue by radinamatic** [learningequality/ka-lite#4910]
## Summary

Seeing some weird new errors on contentrec page when logging in as Learner:

`Unexpected error: 'NoneType' object does not support item assignment`
`TypeError: 'NoneType' object has no attribute '__getitem__'`
## Branch or installer method

Alpha3 deb installer on Ubuntu 0.15

## Snippet from server.log

From 2 different server starts, 
```
[INFO] [2016-02-24 17:31:23,814] cherrypy.access.140456032666064: 127.0.0.1 - - [24/Feb/2016:17:31:23] "GET /static/images/distributed/CC-BY-NC-SA.png HTTP/1.1" 304 - "http://127.0.0.1:8008/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:44.0) Gecko/20100101 Firefox/44.0"
[INFO] [2016-02-24 17:31:23,984] django.request: HTTP Request /securesync/api/user/status/ - Response: 200
[ERROR] [2016-02-24 17:31:24,194] kalite: Error in JSON view: /api/content_recommender
Traceback (most recent call last):
  File "/usr/share/kalite/python-packages/fle_utils/internet/decorators.py", line 26, in api_handle_error_with_json_wrapper_fn
    return handler(request, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/kalite/main/api_views.py", line 98, in content_recommender
    resume_recommendations = [set_bool_flag("resume", rec) for rec in get_resume_recommendations(user, request)] if resume else []
  File "/usr/lib/python2.7/dist-packages/kalite/main/api_views.py", line 94, in set_bool_flag
    rec_dict[flag_name] = True
TypeError: 'NoneType' object does not support item assignment
[INFO] [2016-02-24 17:31:24,195] django.request: HTTP Request /api/content_recommender - Response: 500
[INFO] [2016-02-24 17:32:09,053] django.request: HTTP Request / - Response: 200
[INFO] [2016-02-24 17:32:09,117] django.request: HTTP Request / - Response: 200

```

```
[INFO] [2016-02-24 18:00:52,861] cherrypy.access.140168277471696: 127.0.0.1 - - [24/Feb/2016:18:00:52] "GET /static/fonts/glyphicons-halflings-regular.woff HTTP/1.1" 200 23292 "http://127.0.0.1:8008/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/48.0.2564.116 Chrome/48.0.2564.116 Safari/537.36"
[ERROR] [2016-02-24 18:00:53,068] kalite: Error in JSON view: /api/content_recommender
Traceback (most recent call last):
  File "/usr/share/kalite/python-packages/fle_utils/internet/decorators.py", line 26, in api_handle_error_with_json_wrapper_fn
    return handler(request, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/kalite/main/api_views.py", line 101, in content_recommender
    next_recommendations = [set_bool_flag("next", rec) for rec in get_next_recommendations(user, request)] if next else []
  File "/usr/lib/python2.7/dist-packages/kalite/topic_tools/content_recommendation.py", line 83, in get_next_recommendations
    topic_tree_based_data = get_exercises_from_topics(topic_tree_based_data)
  File "/usr/lib/python2.7/dist-packages/kalite/topic_tools/content_recommendation.py", line 252, in get_exercises_from_topics
    exercises = get_topic_contents(topic_id=topic, kinds=["Exercise"])[:5] #can change this line to allow for more to be returned
TypeError: 'NoneType' object has no attribute '__getitem__'
[INFO] [2016-02-24 18:00:53,152] django.request: HTTP Request /api/content_recommender - Response: 500
[ERROR] [2016-02-24 18:00:54,373] kalite: Error in JSON view: /api/content_recommender
Traceback (most recent call last):
  File "/usr/share/kalite/python-packages/fle_utils/internet/decorators.py", line 26, in api_handle_error_with_json_wrapper_fn
    return handler(request, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/kalite/main/api_views.py", line 101, in content_recommender
    next_recommendations = [set_bool_flag("next", rec) for rec in get_next_recommendations(user, request)] if next else []
  File "/usr/lib/python2.7/dist-packages/kalite/topic_tools/content_recommendation.py", line 82, in get_next_recommendations
    topic_tree_based_data = generate_recommendation_data()[current_subtopic]['related_subtopics'][:settings.TOPIC_RECOMMENDATION_DEPTH]
  File "/usr/lib/python2.7/dist-packages/kalite/topic_tools/content_recommendation.py", line 346, in generate_recommendation_data
    other_neighbors = get_subsequent_neighbors(related, recommendation_data, subtopic)
  File "/usr/lib/python2.7/dist-packages/kalite/topic_tools/content_recommendation.py", line 512, in get_subsequent_neighbors
    elif data[left]['related_subtopics'][1].split(' ')[1] == '4': #if the next left neighbor is at dist 4
IndexError: list index out of range
[INFO] [2016-02-24 18:00:54,373] django.request: HTTP Request /api/content_recommender - Response: 500
[INFO] [2016-02-24 18:00:54,654] django.request: HTTP Request /api/content_recommender - Response: 200
```

## How to reproduce

I was unable to consistently reproduce these errors, although they do seem to appear on first login after a restart. Most of the time contentrec seems to load correctly though... Will continue to dig!

## Screenshots

![contentrec](https://cloud.githubusercontent.com/assets/1457929/13293679/f06bc05a-db20-11e5-92cd-fb231d3e9d49.png)

**Comment by rtibbles** [comment]
Thanks for the report, @radinamatic there should be sufficient traceback to get a handle on what is going on here.

**Comment by radinamatic** [comment]
Been poking on two installations (Alpha4 deb on Ubuntu & build 150 on W7) for a few hours now, and I haven't seen these errors again. So unless you know what could be wrong on a bat, let's go on fixing more pressing issues than this. I'll post again if these creep up again.

**Comment by rtibbles** [comment]
At the very least, we should try to catch these errors to prevent them from happening. There's a little careless indexing going on here.

**Comment by radinamatic** [comment]
Understood, happy catching!

**Comment by rtibbles** [comment]
Fixed in #4923
